crates/fs/private/forest.rs

```diff
@@ -0,0 +1,176 @@
+use std::rc::Rc;
+
+use anyhow::Result;
+use libipld::Cid;
+use log::debug;
+
+use crate::{BlockStore, HashOutput};
+
+use super::{hamt::Hamt, namefilter::Namefilter, Key, PrivateNode, PrivateRef, NONCE_SIZE};
+
+//--------------------------------------------------------------------------------------------------
+// Type Definitions
+//--------------------------------------------------------------------------------------------------
+
+pub type EncryptedPrivateNode = (Option<Vec<u8>>, Cid); // TODO(appcypher): Change to PrivateLink<PrivateNode>.
```

Comment:
Let's also create a follow-up issue for changing this according to the spec, i.e. have the header inline, encrypted into the node and have this link just be a `Cid`.

Reply:
https://github.com/wnfs-wg/rs-wnfs/issues/47